community/kernel/src/main/java/org/neo4j/kernel/impl/api/cursor/TxNodePropertyCursor.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.api.cursor;
+
+import org.neo4j.function.IntSupplier;
+import org.neo4j.kernel.api.cursor.PropertyCursor;
+import org.neo4j.kernel.api.cursor.PropertyValue;
+import org.neo4j.kernel.impl.api.KernelStatement;
+
+/**
+ * Overlays transaction state on a {@link PropertyCursor}.
+ */
+public class TxNodePropertyCursor implements PropertyCursor
+{
+    private KernelStatement statement;
+    private PropertyCursor cursor;
+
+    public TxNodePropertyCursor( KernelStatement statement )
+    {
+        this.statement = statement;
+    }
+
+    public TxNodePropertyCursor init( PropertyCursor cursor)
+    {
+        this.cursor = cursor;
+        return this;
+    }
+
+    @Override
+    public boolean next()
+    {
+        return cursor.next();
+    }
+
+    @Override
+    public boolean seek( IntSupplier intSupplier )
+    {
+        return cursor.seek( intSupplier );
+    }
+
+    @Override
+    public boolean seek( int keyId )
```

Comment:
IntSupplier is from the SeekableCursor (as it needs a T, and int is not a valid T, and we don't want to box to Integer). So to avoid creating IntSupplier's, if the client is able to, there are convenience methods that work with the raw int.